packages/create-subscription/src/__tests__/createSubscription-test.internal.js

```diff
@@ -0,0 +1,420 @@
+        const subscription = source.subscribe(callback);
+        return () => subscription.unsubscribe;
+      },
+    });
+
+    function render(value = 'default') {
+      ReactNoop.yield(value);
+      return null;
+    }
+
+    const observable = createReplaySubject('initial');
+
+    ReactNoop.render(<Subscription source={observable}>{render}</Subscription>);
+    expect(ReactNoop.flush()).toEqual(['initial']);
+    observable.next('updated');
+    expect(ReactNoop.flush()).toEqual(['updated']);
+
+    // Unsetting the subscriber prop should reset subscribed values
+    ReactNoop.render(<Subscription>{render}</Subscription>);
+    expect(ReactNoop.flush()).toEqual(['default']);
+  });
+
+  describe('Promises', () => {
+    it('should support Promises', async () => {
+      const Subscription = createSubscription({
+        getValue: source => undefined,
+        subscribe: (source, callback) => {
+          source.then(value => callback(value), value => callback(value));
+          // (Can't unsubscribe from a Promise)
+          return false;
+        },
+      });
+
+      function render(hasLoaded) {
+        if (hasLoaded === undefined) {
+          ReactNoop.yield('loading');
+        } else {
+          ReactNoop.yield(hasLoaded ? 'finished' : 'failed');
+        }
+        return null;
+      }
+
+      let resolveA, rejectB;
+      const promiseA = new Promise((resolve, reject) => {
+        resolveA = resolve;
+      });
+      const promiseB = new Promise((resolve, reject) => {
+        rejectB = reject;
+      });
+
+      // Test a promise that resolves after render
+      ReactNoop.render(<Subscription source={promiseA}>{render}</Subscription>);
+      expect(ReactNoop.flush()).toEqual(['loading']);
+      resolveA(true);
+      await promiseA;
+      expect(ReactNoop.flush()).toEqual(['finished']);
+
+      // Test a promise that resolves before render
+      // Note that this will require an extra render anyway,
+      // Because there is no way to syncrhonously get a Promise's value
+      rejectB(false);
+      ReactNoop.render(<Subscription source={promiseB}>{render}</Subscription>);
+      expect(ReactNoop.flush()).toEqual(['loading']);
+      await promiseB.catch(() => true);
+      expect(ReactNoop.flush()).toEqual(['failed']);
+    });
+
+    it('should still work if unsubscription is managed incorrectly', async () => {
+      const Subscription = createSubscription({
+        getValue: source => undefined,
+        subscribe: (source, callback) => {
+          source.then(callback);
+          // (Can't unsubscribe from a Promise)
+          return false;
+        },
+      });
+
+      function render(value = 'default') {
+        ReactNoop.yield(value);
+        return null;
+      }
+
+      let resolveA, resolveB;
+      const promiseA = new Promise(resolve => (resolveA = resolve));
+      const promiseB = new Promise(resolve => (resolveB = resolve));
+
+      // Subscribe first to Promise A then Promsie B
```

Comment:
Typo: Promsie -> Promise

("Promsie" sounds like an adorable cartoon used to brainwash kids into learning JavaScript)

Reply:
"Promsie" 😆